### ISSUE TITLE
[Youtube] Download error is silent during subtitle download

### ISSUE THREAD

**Issue by RadicalGDPRist** [yt-dlp/yt-dlp#3163]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.03.08.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Description

Reproduction:

1. Use (old) legit cookies across 2 IPs (residential IPv4 and IPv6)
2. Videos, comments download fine, no "too many requests" errors
3. yt-dlp tries to download subtitles after fetching comments
4. No error, exit code 1

Reason:

I ran into "too many requests", but yt-dlp showed no error at all. Instead it exits silently and gives no error message

Expected:

yt-dlp should print a correct error message

Workaround:

Change cookies (e.g. empty cookie file works again). Looks like there's harsher rate-limiting per cookie identity instance since browser/empty cookies work fine on the same IP.

curl with properly loaded "broken" cookie file returns the automated requests error:
```
curl -c broken-vtt-cookies.txt -b broken-vtt-cookies.txt 'https://www.youtube.com/api/timedtext?v=GgpTBPTd_DA&asr_langs=de%2Cen%2Ces%2Cfr%2Cid%2Cit%2Cja%2Cko%2Cnl%2Cpt%2Cru%2Ctr%2Cvi&caps=asr&xoaf=4&hl=en&ip=0.0.0.0&ipbits=0&expire=1648012028&sparams=ip%2Cipbits%2Cexpire%2Cv%2Casr_langs%2Ccaps%2Cxoaf&signature=<removed>.<removed>&key=yt8&kind=asr&lang=ru&tlang=en&fmt=vtt' --verbose
<html><head><meta http-equiv="content-type" content="text/html; charset=utf-8"/><title>Sorry...</title><style> body { font-family: verdana, arial, sans-serif; background-color: #fff; color: #000; }</style></head><body><div><table><tr><td><b><font face=sans-serif size=10><font color=#4285f4>G</font><font color=#ea4335>o</font><font color=#fbbc05>o</font><font color=#4285f4>g</font><font color=#34a853>l</font><font color=#ea4335>e</font></font></b></td><td style="text-align: left; vertical-align: bottom; padding-bottom: 15px; width: 50%"><div style="border-bottom: 1px solid #dfdfdf;">Sorry...</div></td></tr></table></div><div style="margin-left: 4em;"><h1>We're sorry...</h1><p>... but your computer or network may be sending automated queries. To protect our users, we can't process your request right now.</p></div><div style="margin-left: 4em;">See <a href="https://support.google.com/websearch/answer/86640">Google Help</a> for more information.<br/><br/></div><div style="text-align: center; border-top: 1px solid #dfdfdf;"><a href="https://www.google.com">Google Home</a></div></body></html>
```

You get a delicious cookies for reading this!

### Verbose log

```shell
user@hostname /youtube-dl-folder/test
$ rm *.vtt; ../yt-dlp.exe   --compat-options filename,abort-on-error,no-attach-info-json --write-sub --write-auto-sub --sub-lang en --ignore-config https://www.youtube.com/watch?v=GgpTBPTd_DA -f 139 -v --cookies broken-vtt-cookies.txt; echo $?
rm: cannot remove '*.vtt': No such file or directory
[debug] Command-line config: ['--compat-options', 'filename,abort-on-error,no-attach-info-json', '--write-sub', '--write-auto-sub', '--sub-lang', 'en', '--ignore-config', 'https://www.youtube.com/watch?v=GgpTBPTd_DA', '-f', '139', '-v', '--cookies', 'broken-vtt-cookies.txt']
[debug] Encodings: locale cp1251, fs utf-8, out cp1251 (No ANSI), err cp1251 (No ANSI), pref cp1251
[debug] yt-dlp version 2022.03.08.1 [c0c2c57] (win_exe)
[debug] Compatibility options: no-attach-info-json, filename, abort-on-error
[debug] Python version 3.8.10 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg N-90389-g72bb955625-ffmpeg-windows-build-helpers, ffprobe N-90389-g72bb955625-ffmpeg-windows-build-helpers
[debug] Optional libraries: brotli, Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [youtube] Extracting URL: https://www.youtube.com/watch?v=GgpTBPTd_DA
[youtube] GgpTBPTd_DA: Downloading webpage
[debug] [youtube] Extracted SAPISID cookie
[youtube] GgpTBPTd_DA: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[debug] Downloading subtitles: en
[info] GgpTBPTd_DA: Downloading 1 format(s): 139
[info] Writing video subtitles to: ▒▒▒▒ ▒▒▒▒▒▒▒▒▒ ▒▒▒▒▒▒▒ ▒▒▒▒▒▒▒▒ ▒▒ ▒▒▒▒▒▒▒-GgpTBPTd_DA.en.vtt
[debug] Invoking downloader on "https://www.youtube.com/api/timedtext?v=GgpTBPTd_DA&asr_langs=de%2Cen%2Ces%2Cfr%2Cid%2Cit%2Cja%2Cko%2Cnl%2Cpt%2Cru%2Ctr%2Cvi&caps=asr&xoaf=4&hl=en&ip=0.0.0.0&ipbits=0&expire=1648012057&sparams=ip%2Cipbits%2Cexpire%2Cv%2Casr_langs%2Ccaps%2Cxoaf&signature=<removed>.<removed>&key=yt8&kind=asr&lang=ru&tlang=en&fmt=vtt"
1
```

**Comment by pukkandan** [comment]
It would be very weird to have no error message even with `-v`. But I can't reproduce

> Use (old) legit cookies across 2 IPs (residential IPv4 and IPv6)

wdym? can u explain a bit more?

**Comment by RadicalGDPRist** [comment]
I run yt-dlp in two separate terminal windows on one machine to **download two separate channels using one cookies.txt file**. One instance runs with --source-address <my ipv4> and second --source-address <my ipv6> (aka -4 / -6)

There's indeed no error message else I'd have been able to paste it above. There're no lines printed between "Invoking downloader" and exit code (1). See the command-line I ran:

`$ ../yt-dlp.exe  <bla bla>; echo $?`

**Comment by pukkandan** [comment]
> There's indeed no error message else I'd have been able to paste it above. There're no lines printed between "Invoking downloader" and exit code (1). See the command-line I ran:

Yeah, I got that. The only way I can think of how that could happen is for the python interpreter to crash. But no idea what could have caused that

Considering the cookies contain very sensitive information, I assume you wouldn't want to share the file for debugging

**Comment by RadicalGDPRist** [comment]
If it was really needed, I'd share cookies. But this is essentially a 429 Too Many Requests error, it goes away after a short time.

I will try to get this error on Linux with a system python interpreter to exclude the crash as an option (or modify the code to some non-sense URL to trigger 100% of time). I will update you soon, thanks

**Comment by pukkandan** [comment]
> or modify the code to some non-sense URL to trigger 100% of time

Already tried that

```
❯ yt-dlp --write-auto-sub --skip-down GgpTBPTd_DA -v
[debug] Command-line config: ['--write-auto-sub', '--skip-down', 'GgpTBPTd_DA', '-v']
[debug] Encodings: locale cp65001, fs utf-8, out utf-8, err utf-8, pref cp65001
[debug] yt-dlp version 2022.03.08.1 [c0c2c57d3] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 08ba5f7fd
[debug] Python version 3.10.3 (CPython 64bit) - Windows-10-10.0.22000-SP0
[debug] exe versions: ffmpeg N-105465-gbedb4bac1c-20220206 (fdk,setts), ffprobe N-105465-gbedb4bac1c-20220206, phantomjs 2.1.1
[debug] Optional libraries: brotli, certifi, Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [youtube] Extracting URL: GgpTBPTd_DA
[youtube] GgpTBPTd_DA: Downloading webpage
[youtube] GgpTBPTd_DA: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[debug] Downloading subtitles: en
[debug] Default format spec: bestvideo*+bestaudio/best
[info] GgpTBPTd_DA: Downloading 1 format(s): 248+251
[info] Writing video subtitles to: План вторжения русских варваров на Украину [GgpTBPTd_DA].en.vtt
[debug] Invoking downloader on "https://httpbin.org/status/429"
ERROR: Unable to download video subtitles for 'en': HTTP Error 429: TOO MANY REQUESTS
Traceback (most recent call last):
  File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 3858, in _write_subtitles
    self.dl(sub_filename, sub_copy, subtitle=True)
  File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 2834, in dl
    return fd.download(name, new_info, subtitle)
  File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\downloader\common.py", line 453, in download
    ret = self.real_download(filename, info_dict)
  File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\downloader\http.py", line 378, in real_download
    establish_connection()
  File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\downloader\http.py", line 137, in establish_connection
    ctx.data = self.ydl.urlopen(request)
  File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 3589, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "D:\Programs\scoop\apps\python\current\lib\urllib\request.py", line 525, in open
    response = meth(req, response)
  File "D:\Programs\scoop\apps\python\current\lib\urllib\request.py", line 634, in http_response
    response = self.parent.error(
  File "D:\Programs\scoop\apps\python\current\lib\urllib\request.py", line 563, in error
    return self._call_chain(*args)
  File "D:\Programs\scoop\apps\python\current\lib\urllib\request.py", line 496, in _call_chain
    result = func(*args)
  File "D:\Programs\scoop\apps\python\current\lib\urllib\request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 429: TOO MANY REQUESTS

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 1388, in wrapper
    return func(self, *args, **kwargs)
  File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 1472, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 1528, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 2637, in process_video_result
    self.process_info(new_info)
  File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 2888, in process_info
    sub_files = self._write_subtitles(info_dict, temp_filename)
  File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 3863, in _write_subtitles
    raise DownloadError(f'Unable to download video subtitles for {sub_lang!r}: {err}', err)
yt_dlp.utils.DownloadError: Unable to download video subtitles for 'en': HTTP Error 429: TOO MANY REQUESTS
```

**Comment by NanaMizukiAnother7** [comment]
> > or modify the code to some non-sense URL to trigger 100% of time
> 
> Already tried that
> 
> ```
> ❯ yt-dlp --write-auto-sub --skip-down GgpTBPTd_DA -v
> [debug] Command-line config: ['--write-auto-sub', '--skip-down', 'GgpTBPTd_DA', '-v']
> [debug] Encodings: locale cp65001, fs utf-8, out utf-8, err utf-8, pref cp65001
> [debug] yt-dlp version 2022.03.08.1 [c0c2c57d3] (source)
> [debug] Lazy loading extractors is disabled
> [debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
> [debug] Git HEAD: 08ba5f7fd
> [debug] Python version 3.10.3 (CPython 64bit) - Windows-10-10.0.22000-SP0
> [debug] exe versions: ffmpeg N-105465-gbedb4bac1c-20220206 (fdk,setts), ffprobe N-105465-gbedb4bac1c-20220206, phantomjs 2.1.1
> [debug] Optional libraries: brotli, certifi, Cryptodome, mutagen, sqlite, websockets
> [debug] Proxy map: {}
> [debug] [youtube] Extracting URL: GgpTBPTd_DA
> [youtube] GgpTBPTd_DA: Downloading webpage
> [youtube] GgpTBPTd_DA: Downloading android player API JSON
> [debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
> [debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
> [debug] Downloading subtitles: en
> [debug] Default format spec: bestvideo*+bestaudio/best
> [info] GgpTBPTd_DA: Downloading 1 format(s): 248+251
> [info] Writing video subtitles to: План вторжения русских варваров на Украину [GgpTBPTd_DA].en.vtt
> [debug] Invoking downloader on "https://httpbin.org/status/429"
> ERROR: Unable to download video subtitles for 'en': HTTP Error 429: TOO MANY REQUESTS
> Traceback (most recent call last):
>   File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 3858, in _write_subtitles
>     self.dl(sub_filename, sub_copy, subtitle=True)
>   File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 2834, in dl
>     return fd.download(name, new_info, subtitle)
>   File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\downloader\common.py", line 453, in download
>     ret = self.real_download(filename, info_dict)
>   File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\downloader\http.py", line 378, in real_download
>     establish_connection()
>   File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\downloader\http.py", line 137, in establish_connection
>     ctx.data = self.ydl.urlopen(request)
>   File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 3589, in urlopen
>     return self._opener.open(req, timeout=self._socket_timeout)
>   File "D:\Programs\scoop\apps\python\current\lib\urllib\request.py", line 525, in open
>     response = meth(req, response)
>   File "D:\Programs\scoop\apps\python\current\lib\urllib\request.py", line 634, in http_response
>     response = self.parent.error(
>   File "D:\Programs\scoop\apps\python\current\lib\urllib\request.py", line 563, in error
>     return self._call_chain(*args)
>   File "D:\Programs\scoop\apps\python\current\lib\urllib\request.py", line 496, in _call_chain
>     result = func(*args)
>   File "D:\Programs\scoop\apps\python\current\lib\urllib\request.py", line 643, in http_error_default
>     raise HTTPError(req.full_url, code, msg, hdrs, fp)
> urllib.error.HTTPError: HTTP Error 429: TOO MANY REQUESTS
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 1388, in wrapper
>     return func(self, *args, **kwargs)
>   File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 1472, in __extract_info
>     return self.process_ie_result(ie_result, download, extra_info)
>   File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 1528, in process_ie_result
>     ie_result = self.process_video_result(ie_result, download=download)
>   File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 2637, in process_video_result
>     self.process_info(new_info)
>   File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 2888, in process_info
>     sub_files = self._write_subtitles(info_dict, temp_filename)
>   File "D:\Programs\Source\yt-dlp\yt-dlp\yt_dlp\YoutubeDL.py", line 3863, in _write_subtitles
>     raise DownloadError(f'Unable to download video subtitles for {sub_lang!r}: {err}', err)
> yt_dlp.utils.DownloadError: Unable to download video subtitles for 'en': HTTP Error 429: TOO MANY REQUESTS
> ```

I cannot reproduce here using your command:
`C:\Users\User>yt-dlp --write-auto-sub --skip-down GgpTBPTd_DA -v
[debug] Command-line config: ['--write-auto-sub', '--skip-down', 'GgpTBPTd_DA', '-v']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8 (No ANSI), err utf-8 (No ANSI), pref cp1252
[debug] yt-dlp version 2022.03.08.1 [c0c2c57] (win_exe)
[debug] Python version 3.8.10 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg N-105883-g3706fb8f16-20220308 (setts), ffprobe N-105038-g30322ebe3c-sherpya
[debug] Optional libraries: brotli, Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [youtube] Extracting URL: GgpTBPTd_DA
[youtube] GgpTBPTd_DA: Downloading webpage
[youtube] GgpTBPTd_DA: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx
, tbr, vbr, abr, asr, vext, aext, hasaud, id
[debug] Downloading subtitles: en
[debug] Default format spec: bestvideo*+bestaudio/best
[info] GgpTBPTd_DA: Downloading 1 format(s): 248+251
[info] Writing video subtitles to: План вторжения русских варваров на Украину [GgpTBPTd_DA].en.vtt
[debug] Invoking downloader on "https://www.youtube.com/api/timedtext?v=GgpTBPTd_DA&asr_langs=de%2Cen%2Ces%2Cfr%2Cid%2Cit%2Cja%2Cko%2Cnl%2Cp
t%2Cru%2Ctr%2Cvi&caps=asr&xoaf=4&hl=en&ip=0.0.0.0&ipbits=0&expire=1648017929&sparams=ip%2Cipbits%2Cexpire%2Cv%2Casr_langs%2Ccaps%2Cxoaf&sign
ature=602C44CD0F5656AE60A977F9219E67407A40F2B2.CC8EDCDDBCB0388A20C2FF784429F4C201BEA451&key=yt8&kind=asr&lang=ru&tlang=en&fmt=vtt"
[download] Destination: План вторжения русских варваров на Украину [GgpTBPTd_DA].en.vtt
[download] 100% of 86.28KiB in 00:00`

**Comment by RadicalGDPRist** [comment]
@NanaMizukiAnother7 you have to get a 429 from Youtube. I got it on third run after starting from an empty cookie file:

`while true; do rm *.vtt; ../yt-dlp.exe -6 --compat-options filename,abort-on-error,no-attach-info-json --write-sub --write-auto-sub --all-subs --ignore-config https://www.youtube.com/watch?v=GgpTBPTd_DA -f 139 -v --cookies broken-vtt-cookies; exitcode=$?; echo $exitcode; test $exitcode -gt 0 && break; done`

```
rm: cannot remove '*.vtt': No such file or directory
[debug] Command-line config: ['-6', '--compat-options', 'filename,abort-on-error,no-attach-info-json', '--write-sub', '--write-auto-sub', '--all-subs', '--ignore-config', 'https://www.youtube.com/watch?v=GgpTBPTd_DA', '-f', '139', '-v', '--cookies', 'broken-vtt-cookies']
[debug] Encodings: locale cp1251, fs utf-8, out cp1251 (No ANSI), err cp1251 (No ANSI), pref cp1251
[debug] yt-dlp version 2022.03.08.1 [c0c2c57] (win_exe)
[debug] Compatibility options: abort-on-error, no-attach-info-json, filename
[debug] Python version 3.8.10 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg N-90389-g72bb955625-ffmpeg-windows-build-helpers, ffprobe N-90389-g72bb955625-ffmpeg-windows-build-helpers
[debug] Optional libraries: brotli, Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [youtube] Extracting URL: https://www.youtube.com/watch?v=GgpTBPTd_DA
[youtube] GgpTBPTd_DA: Downloading webpage
[youtube] GgpTBPTd_DA: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[debug] Downloading subtitles: af, sq, am, ar, hy, az, bn, eu, be, bs, bg, my, ca, ceb, zh-Hans, zh-Hant, co, hr, cs, da, nl, en, eo, et, fil, fi, fr, gl, ka, de, el, gu, ht, ha, haw, iw, hi, hmn, hu, is, ig, id, ga, it, ja, jv, kn, kk, km, rw, ko, ku, ky, lo, la, lv, lt, lb, mk, mg, ms, ml, mt, mi, mr, mn, ne, no, ny, or, ps, fa, pl, pt, pa, ro, ru-orig, ru, sm, gd, sr, sn, sd, si, sk, sl, so, st, es, su, sw, sv, tg, ta, tt, te, th, tr, tk, uk, ur, ug, uz, vi, cy, fy, xh, yi, yo, zu
[info] GgpTBPTd_DA: Downloading 1 format(s): 139
[info] Writing video subtitles to: ▒▒▒▒ ▒▒▒▒▒▒▒▒▒ ▒▒▒▒▒▒▒ ▒▒▒▒▒▒▒▒ ▒▒ ▒▒▒▒▒▒▒-GgpTBPTd_DA.af.vtt
[debug] Invoking downloader on "https://www.youtube.com/api/timedtext?v=GgpTBPTd_DA&asr_langs=de%2Cen%2Ces%2Cfr%2Cid%2Cit%2Cja%2Cko%2Cnl%2Cpt%2Cru%2Ctr%2Cvi&caps=asr&xoaf=4&hl=en&ip=0.0.0.0&ipbits=0&expire=1648018476&sparams=ip%2Cipbits%2Cexpire%2Cv%2Casr_langs%2Ccaps%2Cxoaf&signature=<removed>.<removed>&key=yt8&kind=asr&lang=ru&tlang=af&fmt=vtt"
1
```
I will try it on Linux with a fresh Python soon. Pukkadan is probably right. If you want to see what curl gets with these same cookies:

> \* HTTP/2 found, allow multiplexing
< HTTP/2 429
< content-length: 1103

And the "We're sorry..." page

**Comment by RadicalGDPRist** [comment]
**I thought it's my OS but it's not. I can reproduce on Windows 7, Python 3.10 and Manjaro Linux (fresh install).** I downloaded the latest available official release as tar.gz and changed the file to always error (pukkadan's idea):

`yt-dlp\yt_dlp\extractor\youtube.py` -> Line 3477, replace subtitle URL with
`'url': 'https://httpbin.org/status/429',`

Now it will always cause an error. Windows and *fresh* Linux installation, same behavor.

Windows
---

The behavior is the same, Exit code 1 and that's it, using:
- [CPython 3.10.3 from here](https://github.com/indygreg/python-build-standalone/releases/tag/20220318)
- https://github.com/yt-dlp/yt-dlp/releases/tag/2022.03.08.1 (tar.gz version)
-  [api-ms-win-core-path-l1-1-0.dll](https://github.com/nalexandru/api-ms-win-core-path-HACK) for python 3.9+ (last officially supported Python on Windows 7 is 3.8.10). 

```
user@hostname /tempfolder/New Folder (3)/yt-dlp/yt_dlp
$ /tempfolder/cpython-3.10.3+20220318-x86_64-pc-windows-msvc-static-noopt-full/python/install/python.exe __main__.py -6 --compat-options filename,abort-on-error,no-attach-info-json --write-sub --write-auto-sub --all-subs --ignore-config https://www.youtube.com/watch?v=GgpTBPTd_DA -f 139 -v --cookies broken-vtt-cookies; exitcode=$?; echo $exitcode;
[debug] Command-line config: ['-6', '--compat-options', 'filename,abort-on-error,no-attach-info-json', '--write-sub', '--write-auto-sub', '--all-subs', '--ignore-config', 'https://www.youtube.com/watch?v=GgpTBPTd_DA', '-f', '139', '-v', '--cookies', 'broken-vtt-cookies']
[debug] Encodings: locale cp1251, fs utf-8, out cp1251 (No ANSI), err cp1251 (No ANSI), pref cp1251
[debug] yt-dlp version 2022.03.08.1 [c0c2c57d3] (source)
[debug] Compatibility options: filename, abort-on-error, no-attach-info-json
[debug] Python version 3.10.3 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg N-72086-g51f6455 (needs_adtstoasc), ffprobe N-72086-g51f6455
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
[debug] [youtube] Extracting URL: https://www.youtube.com/watch?v=GgpTBPTd_DA
[youtube] GgpTBPTd_DA: Downloading webpage
[youtube] GgpTBPTd_DA: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[debug] Downloading subtitles: af, sq, am, ar, hy, az, bn, eu, be, bs, bg, my, ca, ceb, zh-Hans, zh-Hant, co, hr, cs, da, nl, en, eo, et, fil, fi, fr, gl, ka, de, el, gu, ht, ha, haw, iw, hi, hmn, hu, is, ig, id, ga, it, ja, jv, kn, kk, km, rw, ko, ku, ky, lo, la, lv, lt, lb, mk, mg, ms, ml, mt, mi, mr, mn, ne, no, ny, or, ps, fa, pl, pt, pa, ro, ru-orig, ru, sm, gd, sr, sn, sd, si, sk, sl, so, st, es, su, sw, sv, tg, ta, tt, te, th, tr, tk, uk, ur, ug, uz, vi, cy, fy, xh, yi, yo, zu
[info] GgpTBPTd_DA: Downloading 1 format(s): 139
Deleting existing file ▒▒▒▒ ▒▒▒▒▒▒▒▒▒ ▒▒▒▒▒▒▒ ▒▒▒▒▒▒▒▒ ▒▒ ▒▒▒▒▒▒▒-GgpTBPTd_DA.af.vtt
[info] Writing video subtitles to: ▒▒▒▒ ▒▒▒▒▒▒▒▒▒ ▒▒▒▒▒▒▒ ▒▒▒▒▒▒▒▒ ▒▒ ▒▒▒▒▒▒▒-GgpTBPTd_DA.af.vtt
[debug] Invoking downloader on "https://www.youtube.com/api/timedtext?v=GgpTBPTd_DA&asr_langs=de%2Cen%2Ces%2Cfr%2Cid%2Cit%2Cja%2Cko%2Cnl%2Cpt%2Cru%2Ctr%2Cvi&caps=asr&xoaf=4&hl=en&ip=0.0.0.0&ipbits=0&expire=1648020404&sparams=ip%2Cipbits%2Cexpire%2Cv%2Casr_langs%2Ccaps%2Cxoaf&signature=<removed>.<removed>&key=yt8&kind=asr&lang=ru&tlang=af&fmt=vtt"
1
```

PS: Terminal encoding fix `PYTHONUTF8=1`

Linux, fresh install
----
`5.15.28-1-MANJARO #1 SMP PREEMPT Fri Mar 11 14:12:57 UTC 2022 x86_64 GNU/Linux`
`Python 3.10.2`

Tip: VirtualBox guest additions only work in X session for now (change from Wayland at the bottom left of login screen in KDE)

```
python __main__.py -6 --compat-options filename,abort-on-error,no-attach-info-json --write-sub --write-auto-sub --all-subs --ignore-config "https://www.youtube.com/watch?v=GgpTBPTd_DA" -f 139 -v --cookies broken-vtt-cookies; exitcode=$?; echo $exitcode;
[debug] Command-line config: ['-6', '--compat-options', 'filename,abort-on-error,no-attach-info-json', '--write-sub', '--write-auto-sub', '--all-subs', '--ignore-config', 'https://www.youtube.com/watch?v=GgpTBPTd_DA', '-f', '139', '-v', '--cookies', 'broken-vtt-cookies']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.03.08.1 [c0c2c57d3] (source)
[debug] Compatibility options: abort-on-error, no-attach-info-json, filename
[debug] Python version 3.10.2 (CPython 64bit) - Linux-5.15.28-1-MANJARO-x86_64-with-glibc2.35
[debug] exe versions: ffmpeg 5.0 (setts), ffprobe 5.0, rtmpdump 2.4
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [youtube] Extracting URL: https://www.youtube.com/watch?v=GgpTBPTd_DA
[youtube] GgpTBPTd_DA: Downloading webpage
[youtube] GgpTBPTd_DA: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[debug] Downloading subtitles: af, sq, am, ar, hy, az, bn, eu, be, bs, bg, my, ca, ceb, zh-Hans, zh-Hant, co, hr, cs, da, nl, en, eo, et, fil, fi, fr, gl, ka, de, el, gu, ht, ha, haw, iw, hi, hmn, hu, is, ig, id, ga, it, ja, jv, kn, kk, km, rw, ko, ku, ky, lo, la, lv, lt, lb, mk, mg, ms, ml, mt, mi, mr, mn, ne, no, ny, or, ps, fa, pl, pt, pa, ro, ru-orig, ru, sm, gd, sr, sn, sd, si, sk, sl, so, st, es, su, sw, sv, tg, ta, tt, te, th, tr, tk, uk, ur, ug, uz, vi, cy, fy, xh, yi, yo, zu
[info] GgpTBPTd_DA: Downloading 1 format(s): 139
[info] Writing video subtitles to: План вторжения русских варваров на Украину-GgpTBPTd_DA.af.vtt
[debug] Invoking downloader on "https://httpbin.org/status/429"
1
```
...whats going on?

**Comment by pukkandan** [comment]
What happens if you remove the cookies, but keep the broken URL?

**Comment by RadicalGDPRist** [comment]
Sorry, I hadn't transferred the cookies over. The above comment already has no cookies, the command's been equivalent to:

`python __main__.py -6 --abort-on-error --write-sub --write-auto-sub --all-subs --ignore-config "https://www.youtube.com/watch?v=GgpTBPTd_DA" -f 139 -v; exitcode=$?; echo $exitcode;`

**Comment by pukkandan** [comment]
I can reproduce the issue. I should have been more thorough when trying to reproduce. It is trigerring when a 429 happens with `--abort-on-error`

**Comment by pukkandan** [comment]
Thanks for your help in debugging the issue. If not for you identifying that the issue is not in cookies, this issue would have been (wrongly) stuck as cant-repro

**Comment by RadicalGDPRist** [comment]
:sweat_smile: I shouldn't have named it `broken-vtt-cookies.txt` here. I chose that name locally because only using `curl -c cookies.txt` overwrote my actual cookies with an empty file, and empty cookies no longer got a 429 from yt-dlp (unbreak). If only I had known about the fake URL trick earlier... :)

Thanks for everything and patience! (and sorry about comments.json issue, honestly searched but there were 2 pages of results and I overlooked it in the list)
